linux/net/rina/ipcps/shim-eth-vlan.c

```diff
@@ -1328,18 +1589,38 @@ static int eth_vlan_update_dif_config(struct ipcp_instance_data * data,
                 const struct ipcp_config_entry * entry;
 
                 entry = tmp->entry;
-                if (!strcmp(entry->name,"interface-name")) {
-                        if (!strcpy(info->interface_name,
-                                    entry->value)) {
-                                LOG_ERR("Cannot copy interface name");
-                                return -1;
-                        }
-                } else {
-                        LOG_WARN("Unknown config param for eth shim");
-                        continue;
-                }
+                if (!strcmp(entry->name, "interface-name")) {
+                	ASSERT(entry->value);
+
+                	info->interface_name = rkstrdup(entry->value);
+                	if (!info->interface_name) {
+                		LOG_ERR("Cannot copy interface name");
+                		return -1;
+                	}
+                } else if (!strcmp(entry->name, "qdisc-max-size")) {
```

Comment:
It's optional If you don't do it it uses the defaults, otherwise what is in the template. I've been using this code for the last month, but still not 100% sure on the influence of these values on performance. Therefore it's better that they are configurable for now, to facilitate experimentation with different values.